### PR TITLE
Added Fenwick Tree (Binary Indexed Tree) Algorithm.

### LINKED ISSUE #524: [Feature Request] : Want to add Fenwick Tree (Binary Indexed Tree) Algorithm.
### Is your feature request related to a problem? Please describe.
I'm frustrated when I need to efficiently update or query prefix sums in my project, but there’s no implementation of a Fenwick Tree (Binary Indexed Tree) for such operations.

### Describe the solution you'd like
The addition of Fenwick Tree (BIT) algorithm documentation and code implementation, providing users with an efficient way to perform range queries and point updates. The documentation should include a detailed explanation, time complexity, and a step-by-step guide on its usage.

### Describe alternatives you've considered
Alternatives include using Segment Trees, but Fenwick Trees are often simpler and more efficient for certain use cases like prefix sum and update queries.

### Additional context

Add examples of where this algorithm can be used in competitive programming or real-world scenarios, along with an implementation in the language of choice (e.g., C or JavaScript). Diagrams explaining how the tree works would be helpful.

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by AbhijitMotekar99**
### Description
Added Fenwick Tree (Binary Indexed Tree) Algorithm.

Fixes #524 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Fenwick Tree /FenwickTree.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+---
+id: FenwickTree
```
Comment (ajay-dhangar) [inline docs/Fenwick Tree /FenwickTree.md]: replace `id: FenwickTree` with `id: fenwick-tree`

File: docs/Fenwick Tree /FenwickTree.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+- Query Operation: Accumulates the sum of elements up to a specific index by traversing backward using i -= (i & -i).
+- Range Sum Query: To compute the sum between two indices l and r, the difference of sums up to r and l-1 is used.
+
+## Example Walkthrough
+Suppose we have an array of integers and initialize the Fenwick Tree with zeroes. For each update and query operation, the tree adjusts only the necessary indices, significantly improving efficiency.
+
+## Diagram of Operations
+An example array and its binary representation after a series of updates would visually demonstrate the indices affected by each operation.
+
+![Graph Diagram](./Binary_index.png "Example of Graph for Fenwick tree")
```
Comment (ajay-dhangar) [inline docs/Fenwick Tree /FenwickTree.md]: image not found: Solve issue

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by AbhijitMotekar99** [comment]
Hey @ajay-dhangar 
All changes done !

File: docs/Fenwick Tree /FenwickTree.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+### Key Concepts
+1. **Binary Indexed Representation**: Fenwick Trees leverage binary representation for efficient sum and update operations.
+2. **Operations**:
+   - **Update**: Modify the value of a specific element.
+   - **Query**: Compute the cumulative sum up to a specific index.
+
+### Time Complexity
+- Both update and query operations are `O(log N)` where `N` is the number of elements, due to the logarithmic depth of the tree.
+
+![Fenwick Tree Diagram](https://media.geeksforgeeks.org/wp-content/cdn-uploads/BITSum.png "Example of Fenwick Tree for an Array")
```
Comment (ajay-dhangar) [inline docs/Fenwick Tree /FenwickTree.md]: Not a good format for images.

File: docs/Fenwick Tree /FenwickTree.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+- Query Operation: Accumulates the sum of elements up to a specific index by traversing backward using i -= (i & -i).
+- Range Sum Query: To compute the sum between two indices l and r, the difference of sums up to r and l-1 is used.
+
+## Example Walkthrough
+Suppose we have an array of integers and initialize the Fenwick Tree with zeroes. For each update and query operation, the tree adjusts only the necessary indices, significantly improving efficiency.
+
+## Diagram of Operations
+An example array and its binary representation after a series of updates would visually demonstrate the indices affected by each operation.
+
+![Graph Diagram](./Binary_index.png "Example of Graph for Fenwick tree")
```
Comment (ajay-dhangar) [inline docs/Fenwick Tree /FenwickTree.md]: also, check this

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please follow our contributions guide and also markdown (`.md` /`.mdx`) syntax

**Comment by AbhijitMotekar99** [comment]
Hey @ajay-dhangar 
Now check I resolved all conflicts and remove unnecessary lines.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**